### ISSUE TITLE
sftp connections should use BatchMode=yes by default

### ISSUE THREAD

**Issue by L-B** [lavv17/lftp#564]
Hello,
 
I' m having this issue on RHEL8, which uses lftp 4.8.4.
When using lftp to connect to an sftp:// address, if the host key is not listed yet in ssh_known_hosts, the command will hang indefinitely.
If I add "-o BatchMode=yes", the underlying ssh command behaves properly.
 
Example:
```
$ lftp -u user sftp://localhost
Password:
lftp user@localhost:~> ls
`ls' at 0 [Connecting...]
        <command is hanged here, <Ctrl>+C is needed>
Interrupt                   
lftp user@localhost:~> exit
 
$ lftp -e "set sftp:connect-program \"ssh -a -x -o BatchMode=yes\";" -u user, sftp://localhost
Password:
lftp user@localhost:~> ls
ls: Fatal error: Host key verification failed.
lftp user@localhost:~>
 
$ ssh -a -x
The authenticity of host 'localhost (127.0.0.1)' can't be established.
ECDSA key fingerprint is SHA256:EyzpK1moRazzFirkiIcGIsnH5J7LWOE/VWgZqJ8kC70.
Are you sure you want to continue connecting (yes/no/[fingerprint])?
```

Adding BatchMode would ensure the ssh commands knows it can't act interactively and has to choose a safe fallback.

Thanks!

**Comment by lavv17** [comment]
Should be fixed in 4.9.0

вт, 14 янв. 2020, 12:32 L-B <notifications@github.com>:

**Comment by L-B** [comment]
Ah, https://github.com/lavv17/lftp/commit/0ad0732b8fbacd3519b4e3ecf8c394681b314672 I think! Thanks for the confirmation, I'll pass it along to RH.
